### PR TITLE
docs/website: add anchor links for headings

### LINKED ISSUE #1941: docs/howto: provide anchors for sub-sections of how-to docs
For example, https://gocloud.dev/howto/pubsub/subscribe/ has a section about NATS, but there's no way to link directly to it.

### PR DESCRIPTION
**Pull request by vangent**
I don't really know what I'm doing here but it seems to work :-).

Example:
https://screenshot.googleplex.com/jwFWbFLyCAt

Fixes #1941.

### CODE REVIEW

File: internal/website/layouts/_default/single.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {{ define "main" -}}
 <h1>{{ .Title }}</h1>
-{{ .Content }}
+{{ partial "header-link.html" .Content }}
```
[the remark this answers is not in the document]
Comment (vangent) [reply internal/website/layouts/_default/single.html]: Done.

File: internal/website/layouts/partials/header-link.html [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<!-- https://zwbetz.com/create-header-links-hugo-vs-javascript/ -->
+{{ . | replaceRE "(<h[2-9] id=\"([^\"]+)\">)(.+)(</h[2-9]+>)" `${1} ${3}<a href="#${2}">🔗</a>${4}` | safeHTML }}
```
[the remark this answers is not in the document]
Comment (vangent) [reply internal/website/layouts/partials/header-link.html]: Done.

File: internal/website/static/css/style.css [hunk 3]
```diff
@@ -273,3 +273,29 @@ body {
...
+  text-decoration: none;
+}
+.MainContent h1:hover .anchor,
+.MainContent h2:hover .anchor,
+.MainContent h3:hover .anchor,
+.MainContent h4:hover .anchor,
+.MainContent h5:hover .anchor,
+.MainContent h6:hover .anchor {
+  display: inline;
+}
```
Comment (zombiezen) [inline internal/website/static/css/style.css]: nit: Can you move this up with the rest of the headings styles? Thanks!

Comment (vangent) [reply internal/website/static/css/style.css]: Done.

**Pull request merged**